report/report.md

```diff
@@ -0,0 +1,898 @@
+x \in L , w \in W(x) \}$
+
+A proof of knowledge for relation $R$ with is a two party protocol $(P, V)$
+with the following two properties:
+
+- **Knowledge Completeness:** $Pr[P(w) \iff V_{out} = 1] = 1$, i.e. as in
+  Interactive Proof Systems, after an interaction between the prover and
+  verifier the verifier should be convinced with certainty.  
+- **Knowledge Soundness:** Loosely speaking, in Knowledge Soundness we have
+  an extracter $E$, that when given a possibly malicious prover $P^*$
+  as a turing machine has at least as high as the probability that $P^*$
+  convinces $V$
+
+The above proof systems may be _zero-knowledge_, which in loose terms means
+that anyone looking at the transcript, that is the interaction between
+prover and verifier, will not be able to tell the difference between a real
+transcript and one that is simulated. This means that an adversary will not
+learn any new information, that they could not have gathered themselves. We
+can define the property more formally:
+
+- **Zero-knowledge:** $\forall V^*(\delta). \exists S_{V^*}(x) \in PPT. S_{V^*} \sim^C (P,V^*)$
+
+$V^*$ denotes a prover, honest or otherwise, $\d$ represents information
+that $V^*$ may have from previous executions of the protocol and $(P,V^*)$
+denotes the transcript between the prover and verifier. There is three kinds
+of zero-knowledge:
+
+- **Perfect Zero-knowledge:** $\forall V^*(\delta). \exists S_{V^*}(x) \in PPT. S_{V^*} \sim^P (P,V^*)$,
+  the transcripts $S_{V^*}(x)$ and $(P,V^*)$ are perfectly indistinguishable.
+- **Statistical Zero-knowledge:** $\forall V^*(\delta). \exists S_{V^*}(x) \in PPT. S_{V^*} \sim^S (P,V^*)$,
+  the transcripts $S_{V^*}(x)$ and $(P,V^*)$ are statistically indistinguishable.
+- **Computational Zero-knowledge:** $\forall V^*(\delta). \exists S_{V^*}(x) \in PPT. S_{V^*} \sim^C (P,V^*)$,
+  the transcripts $S_{V^*}(x)$ and $(P,V^*)$ are computationally
+  indistinguishable, i.e. no polynomially bounded adversary $\Ac$ can
+  distinguish them.
+
+### SNARKS
+
+SNARKs - **S**uccinct **N**on-interactive **AR**guments of **K**nowledge
+- have seen increased usage due to their application in blockchains and
+cryptocurrencies. They usually also function as so called general-purpose
+proof schemes. This means that, given any solution to an NP-problem, it will
+produce a proof that the prover knows the solution to said NP-problem. Most
+snarks also allow for zero-knowledge arguments, making them zk-SNARKs.
+
+More concretely, imagine that Alice has todays sudoko problem $x$: She
+claims to have a solution to this problem, her witness, $w$, and want to
+convince Bob without having to send all of it. She could then use a SNARK to
+create a proof to convince Bob. To do this she must first encode the sudoku
+verifier as a circuit $R_x$, and then give the SNARK prover ($\SNARKProver$)
+$R_x(w,w')$ where $w'$ is a public input. This public input could correspond
```

Comment:
Not sure if I understand this! What is w' and why is that necessary when public input can be part of x? We can simply write $R_x(w)$ here (unless I am missing something)

Reply:
It's just notation I suppose. When presented with SNARKs it was presented as above. How about:

```md
To do this she must first encode the sudoku verifier as a circuit $R_x$,
and then give the SNARK prover ($\SNARKProver$) $R_x(w)$. Any public inputs,
such as todays sudoku digit values and positions, can be included as constants 
in the circuit.
```

More explicit/clear, the most important thing is to include that public inputs are handled.